libevmasm/RuleList.h

```diff
@@ -643,6 +643,16 @@ std::vector<SimplificationRule<Pattern>> evmRuleList(
 			Builtins::EXP(2, X),
 			[=]() -> Pattern { return Builtins::SHL(X, 1); }
 		);
+	rules.emplace_back(
```

Comment:
Are you sure this rule will be used? The old optimizer will probably not choose it because the generated code is larger.

Reply:
There is a test for the ir-optimizer. Do you think it won't get applied during the fullsuite?

Reply:
Right, for the new codegen it makes sense, I think. Did you do gas cost tests to see if it is really cheaper than the expanded form?